src/context/playlist-context.js

```diff
@@ -1,11 +1,20 @@
 import axios from "axios";
-import { useState, useEffect, useContext, createContext } from "react";
+import { useReducer, useEffect, useContext, createContext } from "react";
+import { playlistReducer } from "reducer";
 import { useAuth } from "./auth-context";
 
 const PlaylistContext = createContext();
 
 const PlaylistProvider = ({ children }) => {
-  const [playlists, setPlaylists] = useState([]);
+  const [{ playlists, watchlater, likes, history }, dispatch] = useReducer(
+    playlistReducer,
+    {
+      playlists: [],
+      watchlater: [],
+      likes: [],
+      history: [],
+    }
+  );
   const { user } = useAuth();
 
   useEffect(() => {
```

Comment:
I think this might be a better way of writing this if block
```js
if(!user) return;
```

Reply:
the if block is inside this useEffect, github doesn't show it for some reason

Reply:
that would work on initial render but not when user logs in